### PR TITLE
Versioning doc update

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

Please provide a brief description of the changes here.

Update Versioning.md for:
 - Clarity on release policy as we would have a single release package containing SDK, API, and all exporters.
 - Clarity on versioning, as we have a single version for the entire release. 
 - Clarity on ABI compatibility vs ABI breaking across releases.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM.

File: Versioning.md [hunk 1]
```diff
@@ -4,40 +4,35 @@ This document describes the versioning policy for this repository.
...
 
-* ABI stability is guaranteed for the API.
-* ABI stability is not guaranteed for the SDK. In case of ABI breaking changes,
-  instead of bumping up the major version, a new `inline namespace` version will
-  be created, and both old API and new API would be made available
-  simultaneously.
+* The API is header only, and uses ABI compliant interfaces. However, ABI stability
+  is not guaranteed for SDK.
+* In case of ABI breaking changes, a new `inline namespace` version will
+  be introduced, and the existing linked applications can continue using the older version
```
Comment (ThomsonTan) [inline Versioning.md:25]: Could we give an example here on `inline namespace`?

Comment (lalitb) [reply Versioning.md:25]: Sorry seems i missed this comment. ABI versioning using inline namespace is handled here in the code: https://github.com/open-telemetry/opentelemetry-cpp/blob/5767f8e4a2cd32e8ab8647fc5eb672a5e3c21ed6/api/include/opentelemetry/version.h#L13-L14. Should I be adding reference to this ?

Comment (ThomsonTan) [reply Versioning.md:25]: Yes, I think we could add a reference in the next doc update.

**Pull request merged**